src/OmniSharp.Roslyn/HostServicesAggregator.cs

```diff
@@ -34,7 +37,18 @@ public class HostServicesAggregator
             {
                 foreach (var assembly in provider.Assemblies)
                 {
-                    builder.Add(assembly);
+                    try
+                    {
+                        var exportedTypes = assembly.ExportedTypes;
+                        builder.Add(assembly);
+                        logger.LogTrace("Successfully added {assembly} to host service assemblies.", assembly.FullName);
+                    }
+                    catch (Exception)
+                    {
+                        // if we can't see exported types, it means that the assembly cannot participate
+                        // in MefHostServices as one or more of its dependencies (typically a Visual Studio or GACed DLL) is missing
+                        logger.LogWarning("Expected to use {assembly} in host services but the assembly cannot be used due to missing dependencies.", assembly.FullName);
```

Comment:
This message could be misleading since you're just testing for Exception. I can imagine lots of other errors.